### PR TITLE
[fix](nereids) parallel instance number is set to 1 incorrectly

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
make PlanNode.getNumInstance() abstract to force every PlanNode specify how to define its numInstance. 
By default, PlanNode.numInstance is 1. PlanNode except exchangeNode should not use this default value directly. PlanNode.numInstance is used for PlanNode which will change numInstance like exchange node.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morrySnow — COMMENTED** [review]
remove setNumInstances in `PhysicalPlanTranslator`'s method `visitPhysicalWindow` and `visitPhysicalRepeat`

File: regression-test/suites/query_p0/join/test_nestedloop_semi_anti_join.groovy [hunk 1]
```diff
@@ -77,7 +77,6 @@ suite("test_nestedloop_semi_anti_join", "query_p0") {
         select * from ${tbl1} where user_id not in (select user_id from ${tbl2} where ${tbl1}.user_id >  ${tbl2}.user_id) order by ${tbl1}.user_id;
     """
     sql "DROP TABLE IF EXISTS ${tbl1}"
-    sql "DROP TABLE IF EXISTS ${tbl2}"
```
Comment (morrySnow) [inline regression-test/suites/query_p0/join/test_nestedloop_semi_anti_join.groovy:80]: why mod this test case?

Comment (englefly) [reply regression-test/suites/query_p0/join/test_nestedloop_semi_anti_join.groovy:80]: It is not friendly to debug this case.

**Comment by englefly** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.01 seconds
 stream load tsv:          513 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162604797 Bytes

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.22 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         31 seconds loaded 2358488459 Bytes, about 72 MB/s
 stream load orc:          77 seconds loaded 1101869774 Bytes, about 13 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17161993928 Bytes

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**